src/Datadog.Trace.ClrProfiler.Managed/Integrations/MongoDbIntegration.cs

```diff
@@ -0,0 +1,181 @@
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        private static Scope CreateScope(object wireProtocol, object connection)
+        {
+            string databaseName = null;
+            string host = null;
+            string port = null;
+
+            if (wireProtocol.TryGetFieldValue("_databaseNamespace", out object databaseNamespace))
+            {
+                databaseNamespace?.TryGetPropertyValue("DatabaseName", out databaseName);
+            }
+
+            if (connection != null && connection.TryGetPropertyValue("EndPoint", out object endpoint))
+            {
+                if (endpoint is IPEndPoint ipEndPoint)
+                {
+                    host = ipEndPoint.Address.ToString();
+                    port = ipEndPoint.Port.ToString();
+                }
+                else if (endpoint is DnsEndPoint dnsEndPoint)
+                {
+                    host = dnsEndPoint.Host;
+                    port = dnsEndPoint.Port.ToString();
+                }
+            }
+
+            string operationName = null;
+            string collectionName = null;
+            string query = null;
+            string resourceName = null;
+
+            if (wireProtocol.TryGetFieldValue("_command", out object command) && command != null)
+            {
+                try
+                {
+                    // the name of the first element in the command BsonDocument will be the operation type (insert, delete, find, etc)
+                    // and its value is the collection name
+                    if (command.TryCallMethod("GetElement", 0, out object firstElement) && firstElement != null)
+                    {
+                        firstElement.TryGetPropertyValue("Name", out operationName);
+
+                        if (firstElement.TryGetPropertyValue("Value", out object collectionNameObj) && collectionNameObj != null)
+                        {
+                            collectionName = collectionNameObj.ToString();
+                        }
+                    }
+
+                    // get the "query" element from the command BsonDocument, if it exists
+                    if (command.TryCallMethod("Contains", "query", out bool found) && found)
+                    {
+                        if (command.TryCallMethod("GetElement", "query", out object queryElement) && queryElement != null)
+                        {
+                            if (queryElement.TryGetPropertyValue("Value", out object queryValue) && queryValue != null)
+                            {
+                                query = queryValue.ToString();
+                            }
+                        }
+                    }
+                }
+                catch
+                {
+                    // TODO: logging
```

Comment:
Why not do the logging now?

Reply:
Added logging here and a added a few more try/catch with logging as well.